backend/src/run.py

```diff
@@ -164,42 +117,35 @@ def __exit__(self, _exc_type, _exc_value, _exc_traceback):
 @app.route("/nodes")
 async def nodes(_):
     """Gets a list of all nodes as well as the node information"""
-    registry = NodeFactory.get_registry()
-    logger.debug(categories)
+    logger.debug(api.registry.categories)
 
-    # sort nodes in category order
-    sorted_registry = sorted(
-        registry.items(),
-        key=lambda x: category_order.index(NodeFactory.get_node(x[0]).category.name),
-    )
     node_list = []
-    for schema_id, _node_class in sorted_registry:
-        node_object = NodeFactory.get_node(schema_id)
+    for node, sub in api.registry.nodes.values():
         node_dict = {
-            "schemaId": schema_id,
-            "name": node_object.name,
-            "category": node_object.category.name,
-            "inputs": [x.toDict() for x in node_object.inputs],
-            "outputs": [x.toDict() for x in node_object.outputs],
+            "schemaId": node.schema_id,
+            "name": node.name,
+            "category": sub.category.name,
+            "inputs": [x.toDict() for x in node.inputs],
+            "outputs": [x.toDict() for x in node.outputs],
             "groupLayout": [
-                g.toDict() if isinstance(g, Group) else g
-                for g in node_object.group_layout
+                g.toDict() if isinstance(g, Group) else g for g in node.group_layout
             ],
-            "description": node_object.description,
-            "icon": node_object.icon,
-            "subcategory": node_object.sub,
-            "nodeType": node_object.type,
-            "hasSideEffects": node_object.side_effects,
-            "deprecated": node_object.deprecated,
+            "description": node.description,
+            "icon": node.icon,
+            "subcategory": sub.name,
+            "nodeType": node.type,
+            "hasSideEffects": node.side_effects,
+            "deprecated": node.deprecated,
         }
-        if node_object.type == "iterator":
-            node_dict["defaultNodes"] = node_object.get_default_nodes()  # type: ignore
+        if node.type == "iterator":
+            node_dict["defaultNodes"] = node.default_nodes  # type: ignore
```

Comment:
I think we can simplify this if we change the TS type to `readonly defaultNodes?: readonly DefaultNode[] | null;`

```suggestion
        node_dict["defaultNodes"] = node.default_nodes
```